.github/workflows/java-publish.yml

```diff
@@ -99,7 +101,12 @@ jobs:
         run: |
           git config --global user.email "Lance Github Runner"
           git config --global user.name "dev+gha@lancedb.com"
+      - name: Dry run
+        if: github.event_name == 'pull_request'
+        run: |
+          mvn --batch-mode -DskipTests package
       - name: Publish with Java 8
+        if: github.event_name == 'release'
```

Comment:
i am a bit confused here. dont you need to invoke bump version first before running `Publish with Java 8` ?

Reply:
The release pipeline is
- create a github tag with version bumped
- create a github release with the created tag
- a github stable release automatically triggers the java/python publish process

Reply:
ah i see. i've also see you modified the release_process.md. nice. thanks!